docs/developer-guide/rules.md

```diff
@@ -311,8 +311,9 @@ You should:
 
 Deprecating rules doesn't happen very often. When you do, you must:
 
-1. Point the `stylelintReference` link to the specific version of the rule README on the GitHub website, so that it is always accessible.
-2. Add the appropriate metadata to mark the rule as deprecated like `rule.meta = { deprecated: true }`.
+1. Add the appropriate metadata to mark the rule as deprecated like so: `rule.meta = { deprecated: true }`
+2. Set the `stylelintType` to `deprecation`
+3. Optionaly set `stylelintReference` to a link that points to a specific version of the rule's README on your source-code-hosting facility of choice—e.g. GitHub—so that it always remains accessible.
```

Comment:
see https://github.com/stylelint/stylelint/pull/6561#pullrequestreview-1248758992

Reply:
[suggest] How about simplifying the description a bit more and adding an example to help people understand it? E.g.

````suggestion
3. Optionally set `stylelintReference` to a link that points to a specific version of the rule's document so that it always remains accessible.

For example:

```js
result.warn(
  `"your-plugin/old-rule" has been deprecated and will be removed in 7.0. Use '"your-plugin/new-rule" instead.`,
  {
    stylelintType: 'deprecation',
    stylelintReference: 'https://github.com/your-org/your-stylelint-plugin/blob/v6.3.0/src/rules/old-rule/README.md'
  },
);
```
````

Reply:
````suggestion
3. Optionally set `stylelintReference` to a link that points to a specific version of the rule's document so that it always remains accessible.

For example:

```js
result.warn(
  `"your-namespace/old-rule" has been deprecated and will be removed in 7.0. Use '"your-namespace/new-rule" instead.`,
  {
    stylelintType: 'deprecation',
    stylelintReference: 'https://github.com/your-org/your-stylelint-plugin/blob/v6.3.0/src/rules/old-rule/README.md'
  },
);
```
````